harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -597,14 +593,14 @@ def _compute_validation_metrics(self) -> workload.Metrics:
                 num_inputs, num_batches = [sum(n) for n in zip(*input_counts)]
 
         else:
-            check.true(self._evaluate_full_dataset_defined())
+            if not self._evaluate_full_dataset_defined():
+                raise RuntimeError()
             self.validation_loader = cast(torch.utils.data.DataLoader, self.validation_loader)
             if self.is_chief:
                 metrics = self.trial.evaluate_full_dataset(data_loader=self.validation_loader)
 
-                check.is_instance(
-                    metrics, dict, f"eval() must return a dictionary, got {type(metrics)}."
-                )
+                if not isinstance(metrics, dict):
+                    raise TypeError(f"eval() must return a dictionary, got {type(metrics)}.")
```

Comment:
change this to `type(metrics).__name__` which renders much nicer in the output message.

Reply:
Good call. Forgot that.

Reply:
Done.